www/base/src/app/changes/changebuilds/changebuilds.controller.js

```diff
@@ -16,6 +16,12 @@ class ChangeBuildsController {
             $scope.change = change;
         }
 
+        const allChanges = dataAccessor.getChanges();
+
+        allChanges.onChange = (allChanges) => {
+            $scope.last_changeid = allChanges.length;
```

Comment:
this is not the last_changeid, in case there has been some deleted changes.
It is not a good idea to get the list of all changes, can be huge.

Reply:
Needs to use the paging API of the rest API to do this. I am not sure if it is implemented in data-module. You should first implement it there

Reply:
Ok

Reply:
BTW, how can individual changes be deleted? I guess they aren't automatically deleted from bb when removed from the VCS

Reply:
Does paging API mean using `limit`, `offset`, `order` query parameters? If that is the case, then that is already implemented

Reply:
@tardyp

Reply:
What would need to be implemented is a getMore() in the collection class. This would automatically get more data without requerying data that we already have (like it is done now in waterfall's loadMore() function)

Another option is to add the page functionality in the collection class, which would use  limit and offset to change the collection's data to switch to another page (without always having the whole data in memory)
You still need to listen to new data events, because as new data arrive, the page will scroll (some objects will go to previous page, and some objects will come from next page)

Reply:
Ok, thank you